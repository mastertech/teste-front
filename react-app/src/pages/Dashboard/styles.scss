@use "../../themes/variables.scss" as theme;

.card-title {
  margin: theme.$margin-default;
  font-weight: theme.$font-bold;
  @extend .text-lg;
}

.card-content {
  width: 70%;
  display: flex;
  padding: 20px;
  justify-content: space-between;
  transition: all 0.4s;

  @media screen and (max-width: 768px) {
    flex-direction: column-reverse;
    align-items: center;
    text-align: center;
    width: 100%;
  }

  .user-info-text {
    margin: theme.$margin-default;
  }

  .user-avatar {
    border-radius: 50%;
    height: 120px;
    width: 120px;
    overflow: hidden;

    img {
      width: 100%;
    }
  }
}
