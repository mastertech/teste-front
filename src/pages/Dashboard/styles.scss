.dashboard-container {
  margin: 0 auto;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}

.dashboard-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.8rem;
  width:35rem;
  height: 15.5rem;
  background-color: #f5f5f5;
  box-shadow: 0 0 15px rgba(0, 0, 0, 0.3);
 
  button {
    margin-top: 1rem;
  }
}

.dashboard-user-data {
  width: 100%;
  display: flex;
  justify-content: space-between;
  
  h1 {
    margin-top: 0;
    margin-bottom: 2rem;
  }

  p {
    margin-top: 0;
    margin-bottom: 0.5rem;
    color: var(--gray-300);
    font-weight: 500;
  }

  img {
    width: 9rem;
    height: 9rem;
    object-fit: cover;
  }
}

