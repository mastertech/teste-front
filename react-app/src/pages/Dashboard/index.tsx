import { FC, useContext } from 'react'
import { RouteComponentProps } from 'react-router'
import { Button } from '../../components/Button'
import { Card } from '../../components/Card'
import { Title } from '../../components/Title'
import { AuthContext } from '../../context/AuthContext'
import './styles.scss'

type UserProps = {
  birthday: string
  email: string
  gender: string
  id: number
  name: string
  state: string
  avatar: string
}

type DashboardProps = RouteComponentProps<{}, any, { user: UserProps }>

export const Dashboard: FC<DashboardProps> = ({ location }) => {
  const { handleLogout } = useContext(AuthContext)
  const user = location.state?.user

  return (
    <Card>
      {!user && (
        <h1 className="card-title">Sua sess√£o expirou, logue novamente!</h1>
      )}
      <div className="card-content">
        <div className="user-info">
          <Title>{user?.name}</Title>
          <p className="user-info-text">Email: {user?.email}</p>
          <p className="user-info-text">Estado: {user?.state}</p>
        </div>
        <div className="user-avatar">
          <img
            src={
              user?.avatar ||
              'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png'
            }
            alt="User avatar"
            loading="lazy"
          />
        </div>
      </div>
      <Button title="Log out" isLoading={false} onClick={handleLogout} />
    </Card>
  )
}
