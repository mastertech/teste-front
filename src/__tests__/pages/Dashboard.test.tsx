import React from 'react';
import Dashboard from '../../pages/Dashboard';
import renderer from 'react-test-renderer';

describe('Dashboard page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Dashboard />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
