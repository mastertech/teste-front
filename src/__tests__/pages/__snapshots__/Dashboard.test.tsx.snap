// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page renders correctly 1`] = `
<div
  className="dashboard-container"
>
  <main
    className="dashboard-content"
  >
    <div
      className="dashboard-user-data"
    >
      <div>
        <h1 />
        <p>
          Email: 
        </p>
        <p>
          Estado: 
        </p>
      </div>
      <img
        alt="avatar"
      />
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Log out
    </button>
  </main>
</div>
`;
